test/ProtocolRewardsTest.sol

```diff
@@ -0,0 +1,186 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.19;
+
+import "forge-std/Test.sol";
+
+import "../contracts/ProtocolRewards.sol";
+
+contract ProtocolRewardsTest is Test {
+    uint256 internal constant ETH_SUPPLY = 120_200_000 ether;
+
+    ProtocolRewards internal protocolRewards;
+
+    address internal collector;
+    address internal collector2;
+    address internal creator;
+    address internal owner;
+
+    function setUp() public virtual {
+        collector = makeAddr("collector");
+        collector2 = makeAddr("collector2");
+        creator = makeAddr("creator");
+        owner = makeAddr("owner");
+
+        vm.prank(owner);
+        protocolRewards = new ProtocolRewards();
+        vm.label(address(protocolRewards), "protocolRewards");
+
+        vm.deal(collector, 40 ether);
+        vm.deal(creator, 20 ether);
+        vm.deal(owner, 10 ether);
+
+        vm.prank(collector);
+        protocolRewards.deposit{value: 10 ether}(creator, "", "");
+        (bool success,) = address(protocolRewards).call{value: 20 ether}("");
```

Comment:
It'd be good to get the [`expectEmit`](https://book.getfoundry.sh/cheatcodes/expect-emit?highlight=expectEmit#expectemit) in here to check against our events.